references/cli/cluster.go

```diff
@@ -155,6 +155,20 @@ func ensureResourceTrackerCRDInstalled(c client.Client, clusterName string) erro
 	return nil
 }
 
+func ensureVelaSystemNamespaceInstalled(c client.Client, clusterName string) error {
+	ctx := context.Background()
+	remoteCtx := multicluster.ContextWithClusterName(ctx, clusterName)
+	if err := c.Get(remoteCtx, types2.NamespacedName{Name: types.DefaultKubeVelaNS}, &v1.Namespace{}); err != nil {
```

Comment:
I suggest find the KubeVelaNS in management cluster first instead of directly using the default value.

Reply:
How can I find the default namespace in hub cluster? The reason why should create this namespace is our most addons application in vela-system namespace, so the I think this namespace cannot be modified easily.

Reply:
> How can I find the default namespace in hub cluster? The reason why should create this namespace is our most addons application in vela-system namespace, so the I think this namespace cannot be modified easily.

User can set the installed namespace by configuring it while installing helm. See https://github.com/oam-dev/kubevela/blob/3d2fcacb5a98af841b4b69995a0be204f6062c81/charts/vela-core/templates/kubevela-controller.yaml#L81. It is not a hard job.

Therefore, in my opinion, it is necessary to pre-check the installed namespace, since the namespace should be aligned with the kubevela system's namespace.

However, I found that for our addons like [fluxcd](https://github.com/oam-dev/kubevela/blob/3d2fcacb5a98af841b4b69995a0be204f6062c81/charts/vela-core/templates/addons/fluxcd.yaml#L12), the `vela-system` is fixed in the namespace configuration, which I do not think is a well-considered design. In the case where user install vela-core in another namespace, the addons will not work properly. But following the principle that different modules should be decoupled, I prefer not let this problem happens to multi-cluster module again, since sometimes users do not want to use addons together with multi-cluster.

I notice that even finding the namespace of vela-core is not enough since fluxcd addons do not use that. I suggest add another flag `--create-namespace=xxx` to `vela cluster join` to leave an option for users and let the default value to be `vela-system`.